### DISCUSSION TITLE
How to make a jwt-server backend that validates user through sign in page and if user exists redirect back to original url

### DISCUSSION THREAD

**Discussion by kyegomez** [supabase/supabase#10645] [Questions]
I need some guidance on how I would go about validating a user from an external origin and then redirecting them once the user is validated.

- I know JWT's have to be exchanged

the basic diagram would be 

user clicks "Login button" => directed to mysupabaseapp.com/signin/something goes here? => user enters information (email, and password) => if user data exists then redirect to origin url

I know I need JWT's but how do I implement that into the url in the first direct to my supabase app url?
mysupabaseapp.com/signin/something?reference or something

**Comment by kyegomez** [comment]
thanks for your time 💙

**Comment by GaryAustin1** [comment]
Your description is very unclear (at least to me) as to what you are trying to do.
Also for these types of general idea questions, I'd suggest the Supabase Discord server as another avenue for input.
https://discord.com/invite/R7bSpeBSJE

**Reply by kyegomez** [reply]
excuse me.

I'm trying to create a link that directs a user to my webapp where they will sign in and if they have an account redirect back to the original source url. 

I know the link of my webapp has to be modified but what do I add to it?

For example, let's say you download figma and you press signin you are then redirected to a url in the browser where you can log in but when you are veriifed you are redirected back to the original url/external non-browser app
